File: paths.c
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 */

#include <assert.h>
#include <stdio.h>

// constraints
#define MAXN 300001

// input data
int N, M, i;
int X[MAXN], Y[MAXN], A[MAXN], B[MAXN];

int main() {
  //  uncomment the following lines if you want to read/write from files
  //  freopen("input.txt", "r", stdin);
  //  freopen("output.txt", "w", stdout);

  assert(1 == scanf("%d", &N));
  for (i = 0; i < N - 1; i++) assert(2 == scanf("%d %d", &X[i], &Y[i]));
  assert(1 == scanf("%d", &M));
  for (i = 0; i < M; i++) assert(2 == scanf("%d %d", &A[i], &B[i]));

  // insert your code here

  printf("%d\n", 42);  // change 42 with actual answer
  return 0;
}

File: paths.pas
{ constraints }
const
    MAXN = 300001;

{ input data }
var
    N, M, i     : longint;
    X, Y, A, B  : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);
    for i:=0 to N-1 do
        readln(X[i], Y[i]);
    readln(M);
    for i:=0 to M-1 do
        readln(A[i], B[i]);

    { insert your code here }

    writeln(42); { change 42 with actual answer }
end.
